Activate automated release build tests

farodin91 opened the pull request:
Part of #629 and #1628

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

porunov commented:
@farodin91 I am not familiar with GitHub actions. Does this PR moves testing from TravisCI to GitHub actions or it just extends TravisCI builds with GitHub actions?
About those 20000 minutes, do parallel jobs are counted to those minutes or not? I.e. if we have 2 parallel jobs, each job takes 5 minutes to be build. After those jobs are done (in 5 minutes because they are parallel), will it be counted as 10 minutes or 5 minutes?

farodin91 commented:
Does this PR moves testing from TravisCI to GitHub actions? 
I just tried Github actions and added the config into header of PR. Afterwards i removed the changes.

will it be counted as 10 minutes or 5 minutes? If I'm correct it will count 10 minutes.

farodin91 commented:
@porunov This last PR before I'm able to wrap up the cassandra removal.

porunov commented on docs/changelog.md:
```diff
@@ -101,6 +101,10 @@ Parameter `use-deprecated-multitype-index` is no more supported by JanusGraph.
 
+##### Default logging location has changed
+
+If your using `janusgraph.sh` to start your instance, the default logging has been changed from `log` to `logs`
```
`If your using` -> `If you are using`

porunov commented on .travis.yml:
```diff
@@ -176,21 +181,6 @@ jobs:
     - <<: *FULL_BUILD_JOB
       env: MODULE='cql' ARGS=' -Pcassandra3-murmur-ssl -Dcassandra.docker.version=3.0.18 -Dtest=**/diskstorage/cql/CQLStoreTest.java'
 
-    - stage: test
```
Could you please explain why you removed this and what is this?

farodin91 replied:
It was executed one or two year ago the last time. It refers to https://scan.coverity.com/projects/janusgraph-janusgraph.

farodin91 replied:
I will revert these changes and create an issue for it. There are other depending changes.

porunov commented on janusgraph-dist/pom.xml:
```diff
@@ -154,30 +142,72 @@
-                <artifactId>maven-dependency-plugin</artifactId>
+                <groupId>com.googlecode.maven-download-plugin</groupId>
+                <artifactId>download-maven-plugin</artifactId>
+                <version>1.3.0</version>
```
Why not `1.5.0`?

porunov commented on janusgraph-dist/pom.xml:
```diff
@@ -601,41 +593,10 @@
                                         <failIfNoTests>false</failIfNoTests>
                                         <excludes>
                                             <exclude>**/*HBase*IT.java</exclude>
-                                            <exclude>**/*Cassandra*IT.java</exclude>
```
Why did you remove it?

farodin91 replied:
I removed also both classes which are matched by this.

porunov commented on janusgraph-dist/src/test/java/org/janusgraph/pkgtest/CassandraEmbeddedAssemblyIT.java:
```diff
@@ -1,25 +0,0 @@
-// Copyright 2017 JanusGraph Authors
```
Why did you remove it?

farodin91 replied:
Didn't work directly.

porunov commented on janusgraph-dist/src/test/java/org/janusgraph/pkgtest/CassandraEmbeddedESAssemblyIT.java:
```diff
@@ -1,25 +0,0 @@
-// Copyright 2017 JanusGraph Authors
```
Why did you remove it?

farodin91 replied:
Didn't work directly.

farodin91 commented:
I can't verify everything because I am not very much familiar with TravisCI configuration and janusgraph-dist package.
Overall looks good but during review I got a felling that this PR actually is doing several separate things:

- Activates automated release build tests.
> - Removes thrift and cassandra embaded assembly testing and adds cql assembly testing.

They didn't work directly.
> - Makes default ES container to be 7.6.0

I want to be that all elasticsearch default versions are equal.


> I just want to clarify. If we merge this PR, will CQL, Thrift and Cassandraembaded work as expected?
The thing is that we didn't yet remove support for Thrift and Cassandraembaded. As I understand, we will remove it in 0.6.0 release (but we might remove it in 0.5.0 if it doesn't take too much time).

It just removes testing thrift and cassandraembedded from release build.

> As I am not familiar with this code base, I have mostly questions below and a single nitpick

This code base wasn't changed since the fork of titan. The code base is do the most basic tests of an release janusgraph zip.(A lot of people doing these tests, if we vote for a release.) This code base allows combat tests, which currently not working. We could activated it in the near future.
